codegen-client/src/main/kotlin/software/amazon/smithy/rust/codegen/client/smithy/generators/config/StalledStreamProtectionConfigCustomization.kt

```diff
@@ -0,0 +1,169 @@
+                    ) -> Self {
+                        self.set_stalled_stream_protection(#{Some}(stalled_stream_protection_config.into()));
+                        self
+                    }
+                    """,
+                    *codegenScope,
+                )
+
+                rustTemplate(
+                    """
+                    /// Set the [`StalledStreamProtectionConfig`](#{StalledStreamProtectionConfig})
+                    /// to configure protection for stalled streams.
+                    pub fn set_stalled_stream_protection(
+                        &mut self,
+                        stalled_stream_protection_config: #{Option}<#{StalledStreamProtectionConfig}>
+                    ) -> &mut Self {
+                        self.config.store_or_unset(stalled_stream_protection_config);
+                        self
+                    }
+                    """,
+                    *codegenScope,
+                )
+            }
+
+            else -> emptySection
+        }
+    }
+}
+
+class StalledStreamProtectionOperationCustomization(
+    private val codegenContext: ClientCodegenContext,
+) : OperationCustomization() {
+    private val rc = codegenContext.runtimeConfig
+
+    override fun section(section: OperationSection): Writable = writable {
+        when (section) {
+            is OperationSection.AdditionalInterceptors -> {
+                val stalledStreamProtectionModule = RuntimeType.smithyRuntime(rc).resolve("client::stalled_stream_protection")
+                section.registerInterceptor(rc, this) {
+                    // Currently, only response bodies are protected/supported because
+                    // we can't count on hyper to poll a request body on wake.
+                    rustTemplate(
+                        """
+                        #{StalledStreamProtectionInterceptor}::new(#{Kind}::ResponseBody)
+                        """,
+                        *preludeScope,
+                        "StalledStreamProtectionInterceptor" to stalledStreamProtectionModule.resolve("StalledStreamProtectionInterceptor"),
+                        "Kind" to stalledStreamProtectionModule.resolve("StalledStreamProtectionInterceptorKind"),
+                    )
+                }
+            }
+            else -> { }
+        }
+    }
+}
+
+class StalledStreamProtectionRuntimePluginCustomization(codegenContext: ClientCodegenContext) : ServiceRuntimePluginCustomization() {
+    private val rc = codegenContext.runtimeConfig
+
+    override fun section(section: ServiceRuntimePluginSection): Writable = writable {
+        when (section) {
+            is ServiceRuntimePluginSection.AdditionalConfig -> {
+                rustTemplate(
+                    """
+                    ${section.newLayerName}.store_put(
+                        #{StalledStreamProtectionConfig}::new_enabled().grace_period(#{Duration}::from_secs(5)).build()
+                    );
```

Comment:
this is the default right? I think we should be setting this in the `default_runtime_plugin`, right, @jdisanti?

Reply:
We need to make sure this doesn't override what the customer is setting, as well.

Reply:
The `test_stalled_stream_protection_for_downloads_can_be_disabled` ensures that the customer's config isn't overridden.